src/core/sequence_batch_scheduler.cc

```diff
@@ -1150,7 +1128,7 @@ DirectSequenceBatch::SchedulerThread(
             // just use zero for that.
             SetControlTensors(
                 ni, seq_slot, 0 /* corrid */, true /* not_ready */);
-            requests.emplace_back(std::move(ni));
+            curr_payload_->AddRequest(std::move(ni));
```

Comment:
Does AddRequest take a lock or anything similar? The Direct sequence batcher should be able to just form the entire batch ("requests") and then give the whole batch to the RL and not have to update it.

Reply:
No. The `AddRequest` is a function which is specific to Payload class. Only one thread is building the requests for a payload at a time. Hence, no need to lock. The curr_payload_ object exists for each DirectSequenceBatch object. 

The Payload interface was designed in a way to accommodate the requirement of DB where the payload can be grown even after enqueueing to RL. I have kept the interface same here for simplicity.  For DirectSequenceBatch, we update the curr_payload_ as soon as the previous one is enqueued to the RL. Instead of local `std::vector<std::unique_ptr<InferenceRequest>> requests;` we build the request vector directly in the payload.